docs/src/index.md

```diff
@@ -45,11 +45,76 @@ repository of many standard image processing test images.
 
 ## Basic usage
 
-TODO
+Each edge detection algorithm in `ImageEdgeDetection.jl` is an [`AbstractEdgeDetectionAlgorithm`](@ref ImageEdgeDetection.EdgeDetectionAPI.AbstractEdgeDetectionAlgorithm).
 
-```@index
+Suppose one wants to mark the edges in an image. This can be achieved by simply choosing
+an appropriate algorithm and calling [`detect_edges`](@ref) or [`detect_edges!`](@ref) in the
+image.
+
+Let's see a simple demo using the famous Canny edge detection algorithm:
+
+```@example
```

Comment:
FYI, if you add a name after `@example`, then blocks with the same name are shared.

For example, if all example blocks are written `@example canny`, then you only need to `using TestImages, ...; img = testimage("mandril_gray")` once in the first block, and all the contents are still there in the following blocks.

Ref: https://juliadocs.github.io/Documenter.jl/stable/man/syntax/#@example-block

Reply:
Fantastic, thank you.